cmd/snap/cmd_snap_op.go

```diff
@@ -331,6 +331,12 @@ func (mx *channelMixin) setChannelFromCommandline() error {
 		mx.Channel = ch.chName
 	}
 
+	if mx.Channel != "" && mx.Channel != "edge" && mx.Channel != "beta" && mx.Channel != "candidate" && mx.Channel != "stable" {
```

Comment:
unless I'm reading this wrong, this will result in `--channel=foo/bar` ending up with `mx.Channel == "foo/bar/stable"`!